src/util.rs

```diff
@@ -91,36 +90,25 @@ pub(crate) fn format_secret(
 }
 
 pub(crate) fn exec_prompt(
-    conn: zbus::Connection,
+    conn: zbus::blocking::Connection,
     prompt: &ObjectPath,
 ) -> Result<zvariant::OwnedValue> {
-    let prompt_proxy = PromptProxy::new_for(&conn, SS_DBUS_NAME, prompt)?;
+    let prompt_proxy = PromptProxyBlocking::builder(&conn)
+        .destination(SS_DBUS_NAME)?
+        .path(prompt)?
+        .build()?;
 
-    let (tx, rx) = channel();
-
-    // create a handler for `completed` signal
-    prompt_proxy.connect_completed(move |dismissed, result| {
-        let res = if dismissed {
-            Err(Error::Prompt)
-        } else {
-            Ok(result.into())
-        };
-
-        tx.send(res)
-            .expect("rx should not be dropped, channel scoped to this fn");
-
-        Ok(())
-    })?;
+    let mut receive_completed_iter = prompt_proxy.receive_completed()?;
 
     // TODO figure out window_id
     let window_id = "";
     prompt_proxy.prompt(window_id)?;
 
-    // waits for next signal and calls the handler.
-    // If message handled by above handler, `next_signal` returns `Ok(None)`, ending loop.
-    while prompt_proxy.next_signal()?.is_some() {}
-
-    // See comment on allowing channels to panic: https://www.reddit.com/r/rust/comments/awh751/proposal_new_channels_for_rusts_standard_library/ehmk3lz/
-    rx.recv()
-        .expect("tx shold not be dropped, channel scoped to this fn")
+    let signal = receive_completed_iter.next().unwrap();
+    let args = signal.args().unwrap();
```

Comment:
This method returns a Result so maybe `?` would be better here (might need a `.map_error(..)` though)? :thinking: